Mathlib/CategoryTheory/Functor/InvIsos.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+
+! This file was ported from Lean 3 source module category_theory.functor.inv_isos
+! leanprover-community/mathlib commit dc6c365e751e34d100e80fe6e314c3c3e0fd2988
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.EqToHom
+
+/-!
+# Natural isomorphisms with composition with inverses
+
+Definition of useful natural isomorphisms involving inverses of functors.
+These definitions cannot go in `CategoryTheory/Equivalence` because they require `EqToHom`.
+-/
+
+
+namespace CategoryTheory
+
+open CategoryTheory.Functor
+
+universe u₁ u₂ u₃ v₁ v₂ v₃
+
+variable {A : Type u₁} [Category.{v₁} A] {B : Type u₂} [Category.{v₂} B] {C : Type u₃}
+  [Category.{v₃} C]
+
+variable {F : A ⥤ C} {G : A ⥤ B} {H : B ⥤ C}
+
+/-- Construct an isomorphism `F ⋙ H.inv ≅ G` from an isomorphism `F ≅ G ⋙ H`. -/
+@[simps!]
+def compInvIso [h : IsEquivalence H] (i : F ≅ G ⋙ H) : F ⋙ H.inv ≅ G :=
+  isoWhiskerRight i H.inv ≪≫
+    associator G H H.inv ≪≫ isoWhiskerLeft G h.unitIso.symm ≪≫ eqToIso (Functor.comp_id G)
+#align category_theory.comp_inv_iso CategoryTheory.compInvIso
+
+/-- Construct an isomorphism `G ≅ F ⋙ H.inv` from an isomorphism `G ⋙ H ≅ F`. -/
+@[simps!]
+def isoCompInv [_h : IsEquivalence H] (i : G ⋙ H ≅ F) : G ≅ F ⋙ H.inv :=
```

Comment:
just chaning to `_` instead of `_h` does not work here, don't know why (but would like to know)

Reply:
What do you mean with "does not work"?

Reply:
when removing the `h`, I get the following error:
```
expected ']'
```
This error seems unjustified.
In the line that starts with `def isoCompInv` This error seems unjustified.
And then the `Declaration CategoryTheory.isoCompInv not found.`, which makes sense given the first error, I suppose.

Reply:
maybe we should take this discussion to zulip, if it is not trivial to answer? @jcommelin